engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/UpdateByOperatorFactory.java

```diff
@@ -1240,6 +1261,166 @@ private UpdateByOperator makeRollingCountOperator(@NotNull final MatchPair pair,
+                    final String inputColumnName = filterColumnName.get(ii);
+                    final int inputColumnIndex = inputColumnMap.computeIfAbsent(inputColumnName, k -> {
+                        inputColumnNameList.add(inputColumnName);
+                        return inputColumnNameList.size() - 1;
+                    });
+                    inputColumnIndices[ii] = inputColumnIndex;
+                }
+                filterInputColumnIndicesList.add(inputColumnIndices);
+            }
+
+            // Gather the input column type info.
+            final String[] inputColumnNames = inputColumnNameList.toArray(String[]::new);
+            final ChunkType[] inputChunkTypes = new ChunkType[inputColumnNames.length];
+            final Class<?>[] inputColumnTypes = new Class[inputColumnNames.length];
+            final Class<?>[] inputComponentTypes = new Class[inputColumnNames.length];
+            for (int i = 0; i < inputColumnNames.length; i++) {
+                final ColumnDefinition<?> columnDef = tableDef.getColumn(inputColumnNames[i]);
+                inputColumnTypes[i] = columnDef.getDataType();
+                inputChunkTypes[i] = ChunkType.fromElementType(inputColumnTypes[i]);
+                inputComponentTypes[i] = columnDef.getComponentType();
+            }
+
+            // Create a dummy table we can use to initialize filters.
+            final Map<String, ColumnSource<?>> columnSourceMap = new LinkedHashMap<>();
+            for (int i = 0; i < inputColumnNames.length; i++) {
+                final ColumnDefinition<?> columnDef = tableDef.getColumn(inputColumnNames[i]);
+                final ColumnSource<?> source =
+                        NullValueColumnSource.getInstance(columnDef.getDataType(), columnDef.getComponentType());
+                columnSourceMap.put(inputColumnNames[i], source);
+            }
+            final Table dummyTable = new QueryTable(RowSetFactory.empty().toTracking(), columnSourceMap);
+
+            final CountFilter[] countFilters =
+                    CountFilter.createCountFilters(whereFilters, dummyTable, filterInputColumnIndicesList);
+
+            // If any filter is a standard WhereFilter, we need a chunk source table.
+            final boolean chunkSourceTableRequired =
+                    Arrays.asList(countFilters).stream().anyMatch(filter -> filter.whereFilter() != null);
+
+            // Create a new column pair with the same name for the left and right columns
+            final MatchPair pair = new MatchPair(rs.column().name(), rs.column().name());
+
+            return new CountWhereOperator(
+                    pair,
+                    countFilters,
+                    inputColumnNames,
+                    inputChunkTypes,
+                    inputColumnTypes,
+                    inputComponentTypes,
+                    chunkSourceTableRequired);
+        }
+
+        private UpdateByOperator makeRollingCountWhereOperator(
```

Comment:
Lots of identical code with `makeCumCountWhereOperator()`, can almost certainly share code.